plugins/blip/basics/chartbasicsfactory.js

```diff
@@ -89,7 +92,47 @@ var BasicsChart = React.createClass({
       basicsData = _.assign({}, basicsData, _.cloneDeep(basicsState));
       var dataMunger = dataMungerMkr(this.props.bgClasses);
       dataMunger.reduceByDay(basicsData);
-      basicsData.data.reservoirChange.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData);
+
+      // get site changes from three event types
+      basicsData.data.cannulaPrime.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_CANNULA);
+      basicsData.data.tubingPrime.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_TUBING);
+      basicsData.data.reservoirChange.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_RESERVOIR);
+
+      var siteChangeEvents = [];
+
+      if (basicsData.data.tubingPrime.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.tubingPrime.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+      if (basicsData.data.cannulaPrime.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.cannulaPrime.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+      if (basicsData.data.reservoirChange.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.reservoirChange.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+
+      if (siteChangeEvents.length) {
+        // get latest pump upload:
+        var latestPump = _.last(_.sortBy(siteChangeEvents, 'time')).source;
```

Comment:
FYI - There is at least one pair of infusion sets that have interchangable connectors. Specifically, the Tandem T-90 (w/ luer lock) and Medtronic Mio (w/ Medtronic connector). We switch between the Tandem and Medtronic pumps without requiring a site change.